torch/csrc/lazy/core/shape.cpp

```diff
@@ -33,5 +33,15 @@ std::vector<Shape> convertShapes(
   return shape;
 }
 
+size_t Shape::numel() const {
```

Comment:
Wondering if we want to cache the result somewhere? Also, should we consider integer overflow like: https://github.com/pytorch/pytorch/blob/master/c10/core/TensorImpl.h#L2207?

Reply:
Caching could be a good idea, I'm not sure if it's called often enough to warrant the complexity tradeoff, but it seems simple enough given Shape class is small for the time being.

Integer overflow probably isn't a likely scenario for our tensors.  Note the comment they have for the compute_safe_numel version:
`Use only for tensors with sparse layouts because only
   * sparse tensor are likely to have sizes that may lead to integer
   * overflow when computing numel`
   
We don't support sparse-lazy tensors.

Reply:
I have a use case in `LTCTensorImpl` that will set it's `numel` by using this `numel`. So I think caching would be a good idea.

Reply:
hmm.  now you make me wonder (a) still whether caching is actually important for perf (should we measure first?). (b) more importantly, should it be at the Shape layer or at the TensorImpl layer?  I wonder if TensorImpl layer has more visibility into whether there is an actual at::Tensor data or an IR/computation backing the lazy tensor, and caching in LTCTensorImpl would allow taking advantage of that knowledge.

Reply:
`LTCTensorImpl` caches this value by default, it only queries shape when the generation increments. So I guess it's okay for `LTCTensorImpl`. It's true that it's just my gut feeling that caching could be benefit.